akka-sample-sharding-scala/src/main/scala/sample/sharding/WeatherRoutes.scala

```diff
@@ -0,0 +1,44 @@
+package sample.sharding
+
+import java.util.concurrent.TimeUnit
+
+import scala.util.{ Failure, Success }
+import akka.actor.typed.{ ActorRef, ActorSystem }
+import akka.{ actor => classic }
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Route
+import akka.util.Timeout
+
+class WeatherRoutes(guardian: ActorRef[Guardian.Command])(implicit system: ActorSystem[_]) {
+
+  import akka.actor.typed.scaladsl.adapter._
+  implicit val classicSystem: classic.ActorSystem = system.toClassic
+
+  implicit val timeout: Timeout =
+    Timeout(system.settings.config.getDuration("weather.routes.ask-timeout").toMillis, TimeUnit.MILLISECONDS)
+
+  import akka.actor.typed.scaladsl.AskPattern._
+  import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+  import akka.http.scaladsl.server.Directives._
+  import JsonFormats._
+  import Guardian._
+
+  val weather: Route =
+    concat(
+      pathPrefix("temperatures") {
+        post {
+          entity(as[UpdateDevice]) { data =>
+            guardian ! data
+            complete(s"KillrWeather updating data for temperature device ${data.deviceId}")
+          }
+        }
+      },
+      pathPrefix("weather-stations") {
```

Comment:
This would match "weather-stations/a/b/c" as well, I think the right thing here should be `path("weather-stations")`. The quickstart sample used `pathPrefix` since it continued to match the rest of the path nested directives .